server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -403,6 +440,168 @@ public TimeValue timeout() {
+
+                List<String> indices = Arrays.asList(indexNameExpressionResolver.concreteIndexNames(currentState, request));
+
+                final List<String> dataStreams = indexNameExpressionResolver.dataStreamNames(
+                    currentState,
+                    request.indicesOptions(),
+                    request.indices()
+                );
+
+                logger.trace("[{}][{}] creating snapshot for indices [{}]", repositoryName, snapshotName, indices);
+
+                final List<IndexId> indexIds = repositoryData.resolveNewIndices(
+                    indices,
+                    getInFlightIndexIds(runningSnapshots, repositoryName)
+                );
+                final Version version = minCompatibleVersion(currentState.nodes().getMinNodeVersion(), repositoryData, null);
+                final ShardGenerations shardGenerations = buildShardsGenerationFromRepositoryData(
+                    currentState.metadata(),
+                    currentState.routingTable(),
+                    indexIds,
+                    repositoryData
+                );
+
+                boolean remoteStoreIndexShallowCopy = REMOTE_STORE_INDEX_SHALLOW_COPY.get(repository.getMetadata().settings());
+                assert remoteStoreIndexShallowCopy : "remote_store_index_shallow_copy setting is set as false";
+                if (repositoryData.getGenId() == RepositoryData.UNKNOWN_REPO_GEN) {
+                    logger.debug("[{}] was aborted before starting", snapshot);
+                    throw new SnapshotException(snapshot, "Aborted on initialization");
+                }
+                final SnapshotInfo snapshotInfo = new SnapshotInfo(
+                    snapshot.getSnapshotId(),
+                    shardGenerations.indices().stream().map(IndexId::getName).collect(Collectors.toList()),
+                    dataStreams,
+                    pinnedTimestamp,
+                    null,
+                    System.currentTimeMillis(),
+                    shardGenerations.totalShards(),
+                    Collections.emptyList(),
+                    request.includeGlobalState(),
+                    userMeta,
+                    remoteStoreIndexShallowCopy,
+                    pinnedTimestamp
+                );
+                if (!clusterService.state().nodes().isLocalNodeElectedClusterManager()) {
+                    throw new SnapshotException(repositoryName, snapshotName, "Aborting Snapshot, no longer cluster manager");
+                }
+                final StepListener<RepositoryData> pinnedTimestampListener = new StepListener<>();
+                pinnedTimestampListener.whenComplete(repoData -> {
+                    // completeListenersIgnoringException(endAndGetListenersToResolve(snapshot), Tuple.tuple(repoData, snapshotInfo));
+                    listener.onResponse(snapshotInfo);
+                }, listener::onFailure);
+                repository.finalizeSnapshot(
```

Comment:
Will `finalizeSnapshot` update the cluster state?

Reply:
yes it does update repository metadata and generation in the cluster state

Reply:
Do we strictly need the state update?
What is the priority of this cluster state update? Will it impact the sync nature of the API?

Reply:
yeah, we need repository metadata update for backward compatibility. Currently the priority of cluster state update task is normal. I think we can change the priority to `URGENT`, don't see a harm in doing so since its only one state update.

Reply:
Updated cluster update task priority to IMMEDIATE for v2 snapshots

Reply:
Sorry what is the backward compatibility concern since the snapshot itself is different